### PR TITLE
ENH: Support NumPy array-like inputs to image filters

### PR DESCRIPTION
**Pull request by thewtex**
When a NumPy array-like is passed into Python snake_case interface to
filters, the array-like arguments or standard image keyword arguments
are converted into itk.Image views, and passed to the filter. All image
outputs are converted to NumPy ndarray's if array's were used as
inputs.

Addresses #1136

### CODE REVIEW

**Comment by mrocklin** [comment]
Well this is exciting!  I'm glad to see that this is such a small code change for you all.

File: Wrapping/Generators/Python/itkHelpers.py [hunk 1]
```diff
@@ -17,8 +17,60 @@
...
+            if key.lower() in potential_image_input_kwargs and is_arraylike(value):
+                have_array_like_input = True
+                array = np.asarray(value)
+                image = itk.image_view_from_array(array)
+                kwargs[key] = image
+
+        if have_array_like_input:
+            # Convert output itk.Image's to numpy.ndarray's
+            output = image_filter(*tuple(args_list), **kwargs)
+            if output is tuple:
```
Comment (mrocklin) [inline Wrapping/Generators/Python/itkHelpers.py]:
```suggestion
            if isinstance(output, tuple):
```

**Comment by jakirkham** [comment]
Would it be possible also to handle anything that supports the buffer protocol?

**Comment by jakirkham** [comment]
Otherwise this looks great! Thanks for putting it together Matt 🙂

**Comment by thewtex** [comment]
Thanks for the reviews!

Addressed the issue identified by @mrocklin

**Comment by thewtex** [comment]
> Would it be possible also to handle anything that supports the buffer protocol?

The intention is to support anything NumPy array-like, e.g. `dask.array`, with `np.asarray`:

https://github.com/InsightSoftwareConsortium/ITK/pull/1176/files#diff-7404aefb5771301dd2c2edfe74f17008R48

but, it sounds like this could / should change to `np.coerce` in the future following [the discussion of NEP 30](https://mail.python.org/pipermail/numpy-discussion/2019-August/079897.html).

**Comment by mrocklin** [comment]
Woo!

**Pull request merged**